cmd/chartsvc/handler.go

```diff
@@ -200,6 +204,90 @@ func getChartVersionValues(w http.ResponseWriter, req *http.Request, params Para
 	w.Write([]byte(files.Values))
 }
 
+func isChartContained(chartData *models.Chart, chartMetadata *chart.Metadata) bool {
+	eq := true
+	// Compare simple fields
+	if chartData.Description != chartMetadata.Description ||
+		chartData.Home != chartMetadata.Home ||
+		chartData.Icon != chartMetadata.Icon {
+		eq = false
+	}
+	// Compare array of key words (but it may be empty)
+	if len(chartData.Keywords) > 0 && len(chartMetadata.Keywords) > 0 {
+		if !reflect.DeepEqual(chartData.Keywords, chartMetadata.Keywords) {
+			eq = false
+		}
+	}
+	// Compare array of maintainers (but it may be empty)
+	if len(chartData.Maintainers) > 0 && len(chartMetadata.Maintainers) > 0 {
+		// Maintainers format is different (pointers versus structs)
+		c2MaintainerArray := []chart.Maintainer{}
+		for _, m := range chartMetadata.Maintainers {
+			c2MaintainerArray = append(c2MaintainerArray, *m)
+		}
+		if !reflect.DeepEqual(chartData.Maintainers, c2MaintainerArray) {
+			eq = false
+		}
+	}
+	if eq {
+		// Charts matches, look if the version exists
+		for _, v := range chartData.ChartVersions {
+			if v.Version == chartMetadata.Version && v.AppVersion == chartMetadata.AppVersion {
```

Comment:
I am a little bit confused, so this code returns if there is a chart in the repo that maps the one that we are asking about with exactly the same version. Does that mean that we are not returning the actual latest?